### DISCUSSION TITLE
Saving line jumps into the jump list

### DISCUSSION THREAD

**Discussion by adambasis** [JetBrains/ideavim#661] [Q&A] [answered]
My cursor is at line 2. Pressing 3j twice, in sum I jump 6 lines down. So first I was at 2, then 5, now 8. Naturally, if I press `ctrl-o` it doesn't go back to line 5. Can I configure it so, that my line jumps are also saved into the jumplist/navigation list? So when I press `ctrl-o` it goes to line 5, and pressing <c-o> again goes to line 2?

Also, if its relevant, I have these two mapped: `nmap <C-o> <Action>(Back) | nmap <C-i> <Action>(Forward)`

Thank you!

**Comment by citizenmatt** [comment]
You can't configure how the `Back` and `Forward` IDE actions work - they're provided by the IDE. What happens if you remove the mappings? Does IdeaVim's own jumplist functionality provide what you're looking for?

**Comment by adambasis** [comment]
Sadly, no, it doesn't. When jumping lines, it does not get saved into the jumplist.
Is there any way to write an IdeaVim plugin that does that, or anything else? I'd be happy to do so, I just need someone to point me to the right direction. In my Neovim it was configured that when I pressed 2j for example, pressing c-o would bring me back before the 2j command. It was extremely useful, and one of my most used combinations. I feel extremely sluggish without it.

**Reply by citizenmatt** [reply]
Is that default Vim configuration or did you have to set something up to support it? I don't think it's possible to write a plugin to support this, but it's possible to modify IdeaVim's own jump list implementation. It strives to match Vim's own behaviour, so if it's behaving differently, we should take another look.

**Reply by adambasis** [reply]
Yeah, it's not default behavior, it's needed to be set up yourself to have this behavior in vim.
I dont know how beneficial it'd be, but if this would be possible to add to ideavim as a setter or so, that would be awesome.

**Reply by citizenmatt** [reply]
It would be interesting to know what the additional config is to set this up - perhaps this is something that could be supported in IdeaVim.

**Reply by adambasis** [reply]
```
nnoremap <expr> j (v:count <= 1 ? 'j' : "m'" . v:count . 'j')
nnoremap <expr> k (v:count <= 1 ? 'k' : "m'" . v:count . 'k')
xnoremap <expr> j (v:count <= 1 ? 'j' : "m'" . v:count . 'j')
xnoremap <expr> k (v:count <= 1 ? 'k' : "m'" . v:count . 'k')
```

This is my config I use in my NeoVim. When v:count (the number you press before the command i.e. <2>j) is smaller or equal to one, do the normal j command, otherwise save the line jump to the jumplist. The `m'` part of the command does that. so for `2j`, it would do the command `m'2j`.

**Reply by DasOhmoff** [reply]
I am the brother of the author, and I want this feature too! Yes please! Without it, my workflow suffers a lot because going up and down lines is soooo slow. Implementing `v:count` would be a great solution as this is a feature that is extremely versatile and can be used for many things.

**Reply by citizenmatt** [reply, marked as answer]
Have you tried these mappings? We have a test that does what looks to be exactly what you're asking for 😄 

https://github.com/JetBrains/ideavim/blob/9271ca082c1b042fa22e13c2db353b7eb484b50a/src/test/java/org/jetbrains/plugins/ideavim/ex/implementation/services/VimVariableServiceTest.kt#L48-L57

**Reply by adambasis** [reply]
Hahaha, nice! Thanks, this works. Remember to let `<c-o>` and `<c-i>` be handled by IdeaVim. I had them mapped as IDE shortcuts lol, that's why it didn't work :/
